Kudu.Core/Functions/KedaFunctionTriggerProvider.cs

```diff
@@ -150,5 +178,61 @@ public static string GetKedaTriggerType(string triggerType)
                     return triggerType;
             }
         }
+        
+        private static bool IsDurable(FunctionTrigger function) => 
+            function.Type.Equals("orchestrationTrigger", StringComparison.OrdinalIgnoreCase) ||
+            function.Type.Equals("activityTrigger", StringComparison.OrdinalIgnoreCase) ||
+            function.Type.Equals("entityTrigger", StringComparison.OrdinalIgnoreCase);
+
+        private static bool TryGetDurableKedaTrigger(string hostJsonText, out ScaleTrigger scaleTrigger)
+        {
+            scaleTrigger = null;
+            if (string.IsNullOrEmpty(hostJsonText))
+            {
+                return false;
+            }
+
+            JObject hostJson = JObject.Parse(hostJsonText);
+
+            // Reference: https://docs.microsoft.com/azure/azure-functions/durable/durable-functions-bindings#durable-functions-2-0-host-json
+            JObject storageProviderConfig = hostJson.SelectToken("extensions.durableTask.storageProvider") as JObject;
+            string storageType = storageProviderConfig?["type"]?.ToString();
+
+            // Custom storage types are supported starting in Durable Functions v2.4.2
+            if (string.Equals(storageType, "MicrosoftSQL", StringComparison.OrdinalIgnoreCase))
+            {
+                scaleTrigger = new ScaleTrigger
+                {
+                    // MSSQL scaler reference: https://keda.sh/docs/2.2/scalers/mssql/
+                    Type = "mssql",
+                    Metadata = new Dictionary<string, string>
+                    {
+                        ["query"] = "SELECT dt.GetScaleMetric()",
+                        ["targetValue"] = "1", // super-conservative default
+                        ["connectionStringFromEnv"] = storageProviderConfig?["connectionStringName"]?.ToString(),
+                    }
+                };
+            }
+            else
+            {
+                // TODO: Support for the Azure Storage and Netherite backends
+            }
+
+            return scaleTrigger != null;
+        }
+
+        public class FunctionTrigger
```

Comment:
nit: move public class to its own file.

Reply:
The .NET style guidelines say to put top-level public classes in their own file. This is an inner class. However, I went ahead and just made it private. However, this also required me to make the existing `ParseFunctionJson` method, but it looks like there were no external callers so it was a non-breaking change. Let me know if this is a problem.

Reply:
Thanks. I missed this was an inner class.